lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -534,8 +606,8 @@ Point TruncatedDistribution::getSingularities() const
   for (UnsignedInteger i = 0; i < size; ++i)
   {
     const Scalar x = nontruncatedSingularities[i];
-    if (x >= upperBound_) return singularities;
-    if (x > lowerBound_) singularities.add(x);
+    if (x >= bounds_.getUpperBound()[0]) return singularities;
+    if (x > bounds_.getLowerBound()[0]) singularities.add(x);
```

Comment:
As this method is defined only for 1D distributions, it should throw a NotDefinedException if dimension>1, and the upper and lower bounds should be stored as scalars before the loop to speed-up comparisons.